libs/platform/ebpf_epoch.c

```diff
@@ -445,7 +490,23 @@ _ebpf_epoch_get_release_epoch(_Out_ int64_t* release_epoch)
         lock_state = ebpf_lock_lock(&_ebpf_epoch_cpu_table[cpu_id].lock);
 
         // Clear the flush timer flag.
-        _ebpf_epoch_cpu_table[cpu_id].flags.timer_armed = false;
+        _ebpf_set_per_cpu_flag(&_ebpf_epoch_cpu_table[cpu_id], EBPF_EPOCH_PER_CPU_TIMER_ARMED, false);
+
+        if (!ebpf_list_is_empty(&_ebpf_epoch_cpu_table[cpu_id].free_list)) {
+            if (_ebpf_get_per_cpu_flag(&_ebpf_epoch_cpu_table[cpu_id], EBPF_EPOCH_PER_CPU_STALE)) {
+                // Schedule DPC
+                if (!_ebpf_epoch_cpu_table[cpu_id].stale_worker) {
+                    ebpf_allocate_non_preemptible_work_item(
```

Comment:
why not do this at epoch_initiate? Are you avoiding to do this at initiate to save memory?

Reply:
If the code never runs on a CPU, we will never allocate a DPC for it.